### DISCUSSION TITLE
“resolve” event alert subject & alert text arguments.

### DISCUSSION THREAD

**Discussion by zayalaksme** [jertel/elastalert2#376] [Q&A] [answered]
Below is the docuement generated in elastalert index generated by the rule UP service rule (resolve rule for down service rule).  In this UP service document i didnf found any fields regarding service/client information like location, IP, Port etc which are found in Down service docuement.  How do I pass the required fields to UP service docuemnt/rule later to get those details in Resolve/restored mail?

![image](https://user-images.githubusercontent.com/33970128/127397433-d579b0fa-fe27-4c5c-aff1-f45417e39113.png)

Below image is docuement generated by Service Down rule.
![image](https://user-images.githubusercontent.com/33970128/127398072-7b25c47f-7803-44ed-8c35-fbbf085bdd1b.png)

**Comment by nsano-rururu** [comment, marked as answer]
- Does that mean that the information you want is collected in the elasticsearch index and cannot be obtained?
- I think you should also provide the information of the rule file you created as information.

**Reply by zayalaksme** [reply]
DOWN RULE IS
```
es_host: "10.250.10.63"
es_port: 5555
name: DOWNSERVICE
type: any
index: heartbeat-*
include: 
  - message
  - query_key
  - url.full
  - monitor.name
  - monitor.status
  - fields.Client_Name
  - fields.WAN-IP
  - fields.Location
num_events: 04
timeframe:
  seconds: 30
filter:
- terms:
    monitor.status: ["down"]
query_key: url.full
old_query_limit:
  hours: 1
alert:
- "debug"
- "email"
email:
- "xxxxxx@xxxxx.com"
#include: ["fields.Client_Name", "fields.Location", "fields.WAN-IP"]
from_addr: "xxxxx@xxxxx.com"
es_host: "10.250.10.63"
alert_subject: "{} Internet link {} down at {}"
alert_subject_args:
  - fields.Client_Name
  - fields.WAN-IP
  - pretty_ts()
alert_text: "    Location:  {}\n    WAN IP address: {}\n    Status: {}\n    Time Span: {}\n\n\n\n\n\n\n"
alert_text_args:
  - fields.Location
  - fields.WAN-IP
  - monitor.status
  - monitor.timespan
```


UP/Resolve/Restore RULE IS
```
es_host: "10.250.10.63"
es_port: 5555
name: UPSERVICE
type: flatline
index: elastalert*
query_key: "rule_name"
filter:
- query:
    query_string:
      query: "rule_name:\"DOWNSERVICE\" AND matches:>0"
forget_keys: true
timeframe:
  seconds: 20
threshold: 1
alert:
- "debug"
- "email"
old_query_limit:
  hours: 1
email:
- "xxxxxx@xxxxxx.com"
include: ["host", "message", "errormessage"]
from_addr: "xxxxxx@xxxxxx.com"
alert_subject: "{} Internet link {} restored at {}"
lert_subject_args:
  - fields.Client_Name
  - fields.WAN-IP
  - "@timestamp"
alert_text: "    Location:  {}\n    WAN IP address: {}\n    Status: {}\n    Time Span: {}\n\n\n\n\n\n\n"
alert_text_args:
  - fields.Location
  - fields.WAN-IP
  - monitor.status
  - monitor.timespan
```

**Reply by zayalaksme** [reply]
> * Does that mean that the information you want is collected in the elasticsearch index and cannot be obtained?
> * I think you should also provide the information of the rule file you created as information.

I am not seeeing any down event details like IP address, client name and location in the Service up rule generated document in elastalert index.

i recived the service UP mail as below with out any client details.
![image](https://user-images.githubusercontent.com/33970128/127413912-9c4ac37f-6e8d-425b-b0bb-11ca12ff00cc.png)

**Reply by nsano-rururu** [reply]
In UP / Resolve / Restore RULE IS, I refer to the index of "elastalert *", but it does not appear because the following items are not in the index. I think it's the right move.

```
fields.Client_Name
fields.Location
fields.WAN-IP
monitor.status
monitor.timespan
```

**Reply by nsano-rururu** [reply]
You want to add an index item that is generated by elastalert2 and registered as data. There is no such function, so it cannot be done.

**Reply by nsano-rururu** [reply]
Since it is a specification, there is nothing we can do about it.

**Reply by zayalaksme** [reply]
Yes i know those fields are missing in the event doc which generated at UP service got hitted. My question is how do we send these these fields which are in DOWN filter to the event doc generated when UP service got hitted.
The UP Service rule provided in the Documenation didnt transfer the required fields to Elastalert index? I need above fileds the UP service rule generated event doc.

**Reply by nsano-rururu** [reply]
can not. Don't let me say the same thing over and over again.

**Reply by zayalaksme** [reply]
Sorry I didnt see your last message thats why i sent again.
